src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -1157,13 +1141,27 @@ private record OrderedExprInfo(ExprInfo[] infos) { }
 	 * Function parsing
 	 */
 
+  
+	/**
+	 * Attempts to parse {@link SkriptParser#expr} as a function reference.
+	 *
+	 * @param <T> The return type of the function.
+	 * @return A {@link FunctionReference} if a function is found, or {@code null} if none is found.
+	 */
+	public <T> org.skriptlang.skript.common.function.FunctionReference<T> parseFunctionReference() {
+		if (context != ParseContext.DEFAULT && context != ParseContext.EVENT) {
+			return null;
+		}
+
+		return new FunctionReferenceParser(context, flags).parseFunctionReference(expr);
+	}
+ 
 	private final static Pattern FUNCTION_CALL_PATTERN = Pattern.compile("(" + Functions.functionNamePattern + ")\\((.*)\\)");
 
 	/**
-	 * @param types The required return type or null if it is not used (e.g. when calling a void function)
-	 * @return The parsed function, or null if the given expression is not a function call or is an invalid function call (check for an error to differentiate these two)
+	 * @deprecated Use {{@link #parseFunctionReference()}} instead.
 	 */
-	@SuppressWarnings("unchecked")
+	@Deprecated(forRemoval = true, since = "INSERT VERSION")
 	public <T> @Nullable FunctionReference<T> parseFunction(@Nullable Class<? extends T>... types) {
```

Comment:
shouldn't this just call parseFunctionReference internally so we don't have two possible paths? or is that not feasible for some reason?